Package.swift

```diff
@@ -34,6 +35,9 @@ let package = Package(
                     "src/tree.c",
                     "src/query.c"
                 ],
+                cSettings: [
+                    .define("TREE_SITTER_REUSE_ALLOCATOR")
```

Comment:
Why does this macro need to be defined when compiling the core library? I thought this was an optional macro you could define when compiling an external scanner.

Reply:
So it should be used in the CLI and scanners but not in libraries and bindings?

Reply:
You're right, I suggested that to him and that was a mistake on my part. 

The code was deduped so there's effectively only 1 `alloc.h`. The core lib defining this will use the default `ts_current_*` functions which are just wrappers around the libc functions that abort on failure, but you're right in that the CLI is probably the only spot that should define this when compiling the lib. But otherwise, by default malloc/realloc/calloc/free would be used instead of `ts_malloc_default` and friends

Reply:
The `ts_*` memory allocation functions are now behind the same flag in core as in scanners, so these changes retain the previous behaviour (as far as I understood it) while eliminating the need to have 2 `alloc.h` files.

Reply:
I think that we should retain the old behavior without requiring a new command line flag.

As for how we implement that, the `TREE_SITTER_REUSE_ALLOCATOR` macro (or some other macro that is checked in the `ifdef` could be defined in source code, prior to including `alloc.h`.

Reply:
so an idea I had for that is to define an internal flag for this as well, something like `TS_REUSE_ALLOCATOR` and the `TREE_SITTER` variant is the public one. but this would have a check for *either* flag being defined before it's included in any of the c source files, which might feel a little hacky to do this in both source files + the OR'd check in alloc.h. do you have any other ideas?

Reply:
I think the simplest thing may be to just keep the duplicated header. There' not much code in there.

We don't need the `TS_PUBLIC` thing in the header, right? I would think those attributes only need to appear on the *definition* of the symbols (in alloc.c), not on their declarations?